### ISSUE TITLE
test data used for validation

### ISSUE THREAD

**Issue by delip** [fchollet/keras#916]
https://github.com/fchollet/keras/blob/master/examples/mnist_mlp.py#L52

Might want to update your example ;-)

**Comment by fchollet** [comment]
It's fine because the `validation_data` in this example is used strictly for real-time monitoring. It doesn't end up affecting the learned model in any way.

**Comment by delip** [comment]
Only true if the example is used to run once, and never touched again. If I were running this code doing "real-time monitoring" of the logs and then decide to change any of the hyperparams (the script even notes there's room for tuning), then that would be violating the sanctity of the test data. Overfitting to test data can happen in subtle ways.

**Comment by Euphemiasama** [comment]
Is monitoring the validation loss to reduce the learning rate when it reaches a plateau not considered tweaking the parameters ? So it that case aren't we peeking into the test data (validation data is the case of keras) which would eventually lead to overfitting?
